### PR TITLE
Appveyor improvements

### PR DESCRIPTION
**Pull request by scribam**
- The project is built with the Visual Studio solution instead of CMake. This should prevent contributors using Linux (like myself) to introduce a regression on Windows by missing changes in the Visual Studio project files. See #3559 (AppVeyor was green) and #3588 for example.
- The AppVeyor artifact is generated with Qt 5.9.1 (previously Qt 5.8) like for Travis. The version 5.9.1 of Qt is not mandatory, there is no problem to keep the minimal requirement to Qt 5.7 except maybe if we have bugs fixed in a newer version. The new artifact might help to track down some issues in the following case:
  1. A Windows user builds the project himself with an old but supported version (ex: 5.7) of Qt, has a GUI bug and reports it in GitHub
  2. Someone asks him to try with AppVeyor to see if the issue persists
  3. If yes, it is a valid GUI bug (but could be from Qt). Else, the bug was fixed by Qt in a newer version, a workaround could be implemented or the minimal version of Qt could be increased

**Windows users, please test the artifact from AppVeyor! :slightly_smiling_face:**

### CODE REVIEW

**Comment by hcorion** [comment]
Does Appveyor provide Qt 5.9.2?

**Comment by scribam** [comment]
Nope, Qt 5.9.1 is the latest version available (https://www.appveyor.com/docs/build-environment/#qt)

**Comment by hcorion** [comment]
While you're at it, why not change the travis build to 5.9.2: 
https://github.com/RPCS3/rpcs3/blob/master/qt-installer-noninteractive.qs#L35

**Comment by Asinin3** [comment]
Didn't notice any problems, made a fresh install and ran a few games. Not sure if I should be looking for something specific though.

**Comment by scribam** [comment]
@hcorion Wouldn't be better to sync the version between AppVeyor and Travis? Or do you prefer to keep it as up to date as possible?

Thanks @Asinin3, I don't think there is anything specific to look for, a general testing is fine.

**Comment by hcorion** [comment]
Yeah, keep it as up-to-date as possible.

**Comment by scribam** [comment]
Done, Travis uses now Qt 5.9.2.

File: appveyor.yml [hunk 1]
```diff
@@ -1,44 +1,82 @@
...
+
+# scripts to run before build
 before_build:
- - ps: $env:Date="$(git show -s --date=short --format='%ad')"
- - git submodule update --init 3rdparty/ffmpeg 3rdparty/pugixml asmjit 3rdparty/GSL 3rdparty/libpng Vulkan/glslang Vulkan/Vulkan-LoaderAndValidationLayers Utilities/yaml-cpp 3rdparty/cereal 3rdparty/zlib 3rdparty/hidapi 3rdparty/Optional
- - 7z x zlib.7z -aos -oC:\rpcs3\ > null
- - 7z x vulkan.7z -aos -oC:\rpcs3\Vulkan > null
  - set BRANCH=%APPVEYOR_REPO_NAME%/%APPVEYOR_REPO_BRANCH%/#%APPVEYOR_PULL_REQUEST_NUMBER%
  - if "%APPVEYOR_PULL_REQUEST_NUMBER%"=="" (set BRANCH=%APPVEYOR_REPO_NAME%/%APPVEYOR_REPO_BRANCH%)
- - set CXXFLAGS=-DBRANCH=%BRANCH%
```
Comment (AniLeo) [inline appveyor.yml]: You broke experimental warning here

Comment (scribam) [reply appveyor.yml]: Fixed in #3606

**Comment by henriquenj** [comment]
Nice work @scribam , building from CMake on Windows bothered me as well. Sometimes, a developer on Windows would break the Appveyor build because usually his changes are only tested on VS (I did that once). Running VS on Appveyor makes sure that what was tested in the developers machine end up on the Appveyor.

 With this problem solved, there's still a lot of windows-specific hacks on CMake files that makes it overly complex. Wouldn't be better just to deprecate this build and remove CMake on windows altogether? This would make it easier to maintain the files. What do others think?

**Comment by hcorion** [comment]
I think we should keep the CMake windows build around, at the very least for building with mingw64. Perhaps if it becomes enough of a burden down the line, we could look into removing it.

Any AppVeyor specific hacks in the cmake though, I'd be ok with them being removed.

**Comment by danilaml** [comment]
It would be better to deprecate `sln` files and move to CMake projects, since CMake support in various IDEs  (including VS) improves every year. That way both Windows and Linux devs would be working on a unified project.

**Comment by henriquenj** [comment]
Deprecating VS solution files in favor of a single build system is tempting. But in the end I feel that loads of Windows developers simply don't want to mess with CMake just yet. 

@hcorion Ok, I was planning to remove the hacks but let's keep them for the time being. However the CMake build on windows can silently break now that we don't use it anymore on CI.

**Comment by scribam** [comment]
What about using CMake in Visual Studio (https://blogs.msdn.microsoft.com/vcblog/2016/10/05/cmake-support-in-visual-studio/)? We would have a single build system (CMake) and Windows users would still have the possibility to use Visual Studio easily without generating the solution file by command line.

That would also mean to switch to VS 2017 to have the CMake support. Is there any problem which prevent us to move to the lastest version of VS 2017? It is not really clear for me.

**Comment by Asinin3** [comment]
I believe some of the devs didn't want to switch to 2017 but I think this was mostly because they''re not yet familiar with it.

**Pull request merged**